enhancements/security/tls-artifacts-registry.md

```diff
@@ -0,0 +1,188 @@
+to store necessary information about all certificates in the cluster 
+and sets requirements for TLS artifacts metadata.
+
+## Motivation
+
+TLS artifacts registry is useful for several teams in Red Hat (i.e. 
+which component manages a particular certificate) and customers - 
+for instance, helping them to separate platform-managed certificates 
+from user-managed.
+
+### User Stories
+
+- As an Openshift administrator, I want to differentiate 
+  platform-managed TLS artifacts from user-managed.
+- As an Openshift developer, I want to be able to find which team 
+  manages a particular certificate.
+- As a member of OpenShift concerned with the release process 
+  (TRT, dev, staff engineer, maybe even PM), I want to make sure all 
+  managed certificates have sufficient metadata to find the team 
+  responsible for its lifecycle.
+
+### Goals
+
+Create a registry of all managed TLS artifacts, and ensure that no new 
+certificates are added without sufficient metadata.
+Add a test to the conformance suite that verifies the actual cluster TLS 
+artifacts have necessary metadata.
+
+### Non-Goals
+
+## Proposal
+
+* Define a structure for the certificate registry origin repo
+  * Each TLS artifact can be either
+    * on-disk file
+    * in-cluster secret or config map
+  * Additional expected metadata stored for this artifact
+    * JIRA component
+    * TLS artifact description (one sentence describing the purpose 
+      of this artifact)
+  * Each TLS artifact is associated with a list of owners, responsible 
+    for the lifecycle of this artifact
+* Create a new test, which ensures that all secrets are present in the 
+  registry
+  * Some secrets may be versioned, contain hash or IP address / DNS 
+    address - the test should strip this information and match it to 
+    the canonical location (i.e. 
+    `etcd-peer-ip-10-10-10-10.us-east-1.internal` becomes 
+    `etcd-peer-for-master-2` etc.)
+* Create a test that verifies that TLS artifacts in a cluster have 
+  necessary annotations to provide required metadata.
+* TLS artifact registry is being composed as a JSON file from several 
+  parts (usually one file per a set of owners).
+* In the `origin` repo create `hack/update-tls-ownership.json` and 
+  `hack/verify-tls-ownership.json` scripts to compose TLS artifact 
+  registry JSON from several JSONs and verify that all necessary 
+  metadata is set respectively.
+* Update library-go functions to set metadata when the TLS artifact is 
+  being generated, so that most operators would use the recommended 
+  method.
+
+### Workflow Description
+
+The cluster administrator opens the TLS artifacts registry in the `origin` repo 
+and consults it to ensure that a particular certificate is platform-managed. The administrator can use JIRA component metadata to file a 
+new issue related to this certificate.
+
+Similarly, TRT members use this registry to file issues found in 
```

Comment:
probably not TRT.

Can you describe the workflow for adding a new cert?  I think it looks like
1. open PR adding new cert to operator repo
2. CI fails due to new cert appearing
3. CI includes the new raw-tls info required, but it's still missing an owner
4. dev takes the file and adds it to origin PR
5. origin PR fails CI because he forgot to update
6. dev runs the hack/update
7. origin PR fails CI because dev doesn't have an owner (or other annotation like "works with recert tool) for his new cert
8. dev fakes in his origin PR, thus taking ownership.  OR
9. dev goes to his operator PR and adds the necessary annotations, redoes his origin, PR
10. origin PR now passes CI and merges
11. operator PR can now land.

Reply:
>6. dev runs the hack/update
would fail if owner is unset, so step 7 won't happen.

I expect the happy path to look like this:

1. Developer is aware of this enhancement, so they create a PR with necessary annotation
2. PR merges and TLS test in origin shows a flaking test - unregistered certificate
3. Dev creates a PR to origin, updates ownership, `hack/update` generates new expected raw TLS info file
4. After PR is merged TLS info test no longer flakes

Most unhappy path:
1. open PR adding new cert to operator repo
2. CI fails due to new cert appearing
3. CI includes the new raw-tls info required, but it's still missing an owner
4. dev takes the file and adds it to origin PR
5. origin PR fails CI because he forgot to update
6. dev runs the hack/update. `hack/validate-tls-info` requires owner to be set
7. dev updates the owner thus taking ownership, origin PR is approved, operator PR can now land

Reply:
However, TRT may be interested in this too, as if they encounter a problem with certificate its easy to find the team owning it - and quickly find a bug before reverting the offending PR (if available)

Reply:
I wonder if adding a new cert is considered a failure based on the scenario that David described? Adding a new cert that meets all requirements (such as valid annotations, ownership info and proper expiration time) should be fine.